### PR TITLE
Fixes #16988: Add documentation on generic methods writing

### PR DESCRIPTION
**Pull request by Fdall**
https://issues.rudder.io/issues/16988

### CODE REVIEW

File: src/rudder-by-example/modules/ROOT/pages/exntending-rudder/add-new-methods.adoc [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+For this example we will want to create a method that let us define a
+Rudder variable based on the result of a template.
+
+To do this we will use the pre-existing Mustache templating methods furnished in Rudder:
+* xref:reference:reference:generic_methods.adoc#file_from_template_mustache[Mustache]: it is Rudder's native
+templating engine, with a very simple syntax, and the best performance. It should be the default choice for most cases.
+
+== Prerequisites
+
+All our Linux methods are built on the same manner, using the underlying tool CF-Engine.
```
Comment (amousset) [inline src/rudder-by-example/modules/ROOT/pages/exntending-rudder/add-new-methods.adoc]: CFEngine

Comment (VinceMacBuche) [reply src/rudder-by-example/modules/ROOT/pages/exntending-rudder/add-new-methods.adoc]: missplay

**Comment by Fdall** [comment]
PR updated with a new commit

File: src/rudder-by-example/modules/ROOT/pages/exntending-rudder/add-new-methods.adoc [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+
+To do this we will use the pre-existing Mustache templating methods furnished in Rudder:
+* xref:reference:reference:generic_methods.adoc#file_from_template_mustache[Mustache]: it is Rudder's native
+templating engine, with a very simple syntax, and the best performance. It should be the default choice for most cases.
+
+== Prerequisites
+
+All our Linux methods are built on the same manner, using the underlying tool CFEngine.
+Some basic knowledge on CFEngine will be required to fully understand this guide.
+In rudder we are using a slighty different vocabulary than CFEngine:
```
Comment (amousset) [inline src/rudder-by-example/modules/ROOT/pages/exntending-rudder/add-new-methods.adoc]: Rudder

File: src/rudder-by-example/modules/ROOT/pages/exntending-rudder/add-new-methods.adoc [hunk 3]
```diff
@@ -0,0 +1,233 @@
...
+
+* The `classes` are called `conditions` in Rudder
+* The `bundles` are still called `bundle` but may also describe a proper `method`
```
Comment (amousset) [inline src/rudder-by-example/modules/ROOT/pages/exntending-rudder/add-new-methods.adoc]: The technique id is the bundle name.

Comment (Fdall) [reply src/rudder-by-example/modules/ROOT/pages/exntending-rudder/add-new-methods.adoc]: method id maybe?

**Comment by Fdall** [comment]
PR updated with a new commit

**Comment by Fdall** [comment]
PR updated with a new commit

**Comment by Fdall** [comment]
OK, squash merging this PR

**Pull request merged**